python/ray/serve/_private/application_state.py

```diff
@@ -1034,17 +1038,25 @@ def override_deployment_info(
     for options in deployment_override_options:
         deployment_name = options["name"]
         info = deployment_infos[deployment_name]
+        original_options = info.deployment_config.dict()
+        original_options["user_configured_option_names"].update(set(options))
 
-        if (
-            info.deployment_config.autoscaling_config is not None
-            and info.deployment_config.max_concurrent_queries
-            < info.deployment_config.autoscaling_config.target_num_ongoing_requests_per_replica  # noqa: E501
-        ):
-            logger.warning(
-                "Autoscaling will never happen, "
-                "because 'max_concurrent_queries' is less than "
-                "'target_num_ongoing_requests_per_replica' now."
+        if options.get("num_replicas") == "auto":
+            options["num_replicas"] = None
+            if (
+                "max_concurrent_queries"
+                not in original_options["user_configured_option_names"]
+            ):
+                options["max_concurrent_queries"] = NEW_DEFAULT_MAX_CONCURRENT_QUERIES
+
+            autoscaling_config = (
```

Comment:
leave comment for what this is doing pls

Reply:
Added a comment!